iseq.c

```diff
@@ -1332,17 +1332,33 @@ rb_iseqw_to_iseq(VALUE iseqw)
 
 /*
  *  call-seq:
- *     iseq.eval -> obj
+ *     iseq.eval([binding]) -> obj
  *
  *  Evaluates the instruction sequence and returns the result.
  *
  *      RubyVM::InstructionSequence.compile("1 + 2").eval #=> 3
+ *
+ *  If <em>binding</em> is given, which must be a Binding object, the
+ *  evaluation is performed in its context.
+ *
+ *      obj = Struct.new(:a, :b).new(1, 2)
+ *      bind = obj.instance_eval {binding}
+ *      RubyVM::InstructionSequence.compile("a + b").eval(bind) #=> 3
  */
 static VALUE
-iseqw_eval(VALUE self)
+iseqw_eval(int argc, const VALUE *argv, VALUE self)
 {
-    rb_secure(1);
-    return rb_iseq_eval(iseqw_check(self));
+    VALUE scope;
+
+    if (argc == 0) {
+        rb_secure(1);
+        return rb_iseq_eval(iseqw_check(self));
+    }
+    else {
+        rb_scan_args(argc, argv, "01", &scope);
```

Comment:
With `rb_check_arity(argc, 0, 1)` then `rb_secure(1)`, duplicate code can be reduced.

Reply:
Yes I agree, I'll update it to scan check for the arity and refactor it accordingly